src/cli.rs

```diff
@@ -152,6 +152,26 @@ pub struct RootOpts {
     #[arg(short, long, env = "VECTOR_WATCH_CONFIG")]
     pub watch_config: bool,
 
+    /// Method for configuration watching.
+    ///
+    /// By default, `vector` uses recommended watcher for host OS
+    /// - `inotify` for Linux-based systems.
+    /// - `kqueue` for unix/macos
+    /// - `ReadDirectoryChangesWatcher` for windows
+    /// The `poll` watcher can be used in cases where `inotify` doesn't work, e.g., when attaching the configuration via NFS.
+    #[arg(long, default_value = "inotify", env = "VECTOR_WATCH_METHOD")]
```

Comment:
The env var name is inconsistent with the `cli.cue` here. 
```suggestion
    #[arg(long, default_value = "inotify", env = "VECTOR_WATCH_CONFIG_METHOD")]
```